dbms/src/Storages/Page/V2/tests/gtest_page_storage.cpp

```diff
@@ -727,6 +736,24 @@ try
 }
 CATCH
 
+TEST_F(PageStorage_test, getMaxIdsFromRestore)
+try
+{
+    {
+        WriteBatch batch;
+        batch.putExternal(1, 0);
+        batch.putExternal(2, 0);
+        batch.delPage(1);
+        batch.delPage(2);
+        storage->write(std::move(batch));
+    }
+
+    auto [page_storage, max_ids] = reopen();
```

Comment:
```suggestion
    storage = nullptr;
    auto [page_storage, max_ids] = reopen();
```
reset the opened storage before reopen